projects/wallet-lib-client-for-event-queue/0002-connection-manager-design.md

```diff
@@ -0,0 +1,174 @@
+# Summary
+[summary]: #summary
+
+Organize the connection manager classes and move responsabilities from the wallet facade to the connection manager.
+
+# Motivation
+[motivation]: #motivation
+
+The connection manager should be responsible for managing the state of the
+connection and how the events from the connection are used, this way we can have
+a multiple connection classes to handle the different behaviors but maintaining
+a common interface.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+## Current implementation
+
+The current `Connection` class implementation is a wrapper around a `WebSocket`
+class, the `WebSocket` class is responsible for managing the connection and
+actually sending and receiving data.
+The `Connection` class will interpret the events from the websocket instance and
+emit events that are used by the facade, it also subscribes to specific events
+and handles them differently depending on the type of event.
+
+The layers of abstraction make the connection very easy to instantiate and use
+but create a black box of events that are interpreted by the facade.
+
+## New connection classes
+
+### FullnodePubSubConnection and FullnodeEventQueueWSConnection
+
+Will connect to the fullnode websocket and subscribe to the apropriate
+events, meaning the best block updates and the transaction events.
+
+Transaction events will be inserted into the storage and the history will be
+processed (calculating metadata, balance, etc) by the connection instance.
+Best block updates will be processed as usual.
+
+#### Events
+
+- `new-tx` and `update-tx`
```

Comment:
Semi-unrelated to line number, but remember to add a ping/pong mechanism

Reply:
The document of the event queue and the implementation did not have ping/pong anywhere but from what I understand it comes with the websocket lib we use, so I'll add the ping/pong to the design